### PR TITLE
Add block offset to Push Constant BlockVariable

### PR DESCRIPTION
**Pull request by spencer-lunarg**
A Push Constant block like

```glsl
layout(push_constant) uniform uPushConstant {
    layout(offset = 8) vec2 a;
    float b; // offset 16
    layout(offset = 4) float c;
} pc;
```

should return an offset of `4` instead of `0`

Unlike descriptors, the Push Constant `Block` has a concept of a base offset since

### CODE REVIEW

File: tests/push_constants/non_zero_block_offset.spv.yaml [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+      scalar: { width: 32, signedness: 0 }
+      vector: { component_count: 0 }
+      matrix: { column_count: 0, row_count: 0, stride: 0 }
+    array: { dims_count: 0, dims: [], stride: 0 }
+    member_count: 0
+    members:
+    type_description: *td2
+  - &bv3
+    name: "pc"
+    offset: 4
```
Comment (spencer-lunarg) [inline tests/push_constants/non_zero_block_offset.spv.yaml:212]: was `0` before

File: spirv_reflect.c [hunk 2]
```diff
@@ -3504,6 +3500,14 @@ static SpvReflectResult ParsePushConstantBlocks(
       return result;
     }
 
+    // Get minimum offset for whole Push Constant block
+    // It is not valid SPIR-V to have an empty Push Constant Block
+    p_push_constant->offset = UINT32_MAX;
+    for (uint32_t i = 0; i < p_push_constant->member_count; ++i) {
```
Comment (chaoticbob) [inline spirv_reflect.c:3506]: Maybe pick another index var name? VS is error out on this since `i` is shadowing an existing variable also called `i`.

**Review by chaoticbob — CHANGES_REQUESTED** [review]

**Comment by chaoticbob** [comment]
VS is complaining about shadowing in a few places:
```
spirv_reflect.c(3507,19): error C2220: the following warning is treated as an error [D:\a\SPIRV-Reflect\SPIRV-Reflect\build\spirv-reflect-pp.vcxproj]
spirv_reflect.c(3507,19): warning C4456: declaration of 'i' hides previous local declaration [D:\a\SPIRV-Reflect\SPIRV-Reflect\build\spirv-reflect-pp.vcxproj]
spirv_reflect.c(3445,15): message : see declaration of 'i' [D:\a\SPIRV-Reflect\SPIRV-Reflect\build\spirv-reflect-pp.vcxproj]
spirv_reflect.c(3507,19): error C2220: the following warning is treated as an error [D:\a\SPIRV-Reflect\SPIRV-Reflect\build\spirv-reflect.vcxproj]
spirv_reflect.c(3507,19): warning C4456: declaration of 'i' hides previous local declaration [D:\a\SPIRV-Reflect\SPIRV-Reflect\build\spirv-reflect.vcxproj]
spirv_reflect.c(3445,15): message : see declaration of 'i' [D:\a\SPIRV-Reflect\SPIRV-Reflect\build\spirv-reflect.vcxproj]
```

File: spirv_reflect.c (hunk 2 above)
Comment (spencer-lunarg) [reply spirv_reflect.c:3506]: ...... I fixed this, just forgot to save and commit amend it seems

**Review by chaoticbob — APPROVED** [review]

**Pull request merged**